### ISSUE TITLE
No templates ValueError for HOH when creating System with solvated pdb

### ISSUE THREAD

**Issue by pengtb** [openmm/openmm#3393]
Hi there,
I 'm new to openmm, and I met an error when using solvated pdb to create system. Here are the codes:

***first using modeller to add hydrogens & solvents, then save those to a pdb file***
`ff = forcefield.ForceField('amber/protein.ff14SB.xml', 'amber/tip3p_standard.xml', 'amber/tip3p_HFE_multivalent.xml')`
`pdb = PDBFile(pdbfile)` 
`modeller = Modeller(pdb.topology, pdb.positions)`
`changed = modeller.addHydrogens(ff)`
`modeller.addSolvent(ff, padding=1*nanometer, positiveIon='K+', ionicStrength=0.1*molar)`
`pdb.writeFile(modeller.topology, modeller.positions, open(outputfile, 'w'))`

***then re-load the pdb file and use it to create system***
`pdb = PDBFile(outputfile)`
`system = ff.createSystem(pdb.topology, nonbondedMethod=PME, nonbondedCutoff=1*nanometer, constraints=None)`

***here's the error***
```
ValueError                                Traceback (most recent call last)
/tmp/ipykernel_9253/1888280762.py in <module>
      1 # energy minimize
----> 2 system = ff.createSystem(pdb.topology, nonbondedMethod=PME,
      3         nonbondedCutoff=1*nanometer, constraints=None)
      4 integrator = LangevinMiddleIntegrator(300*kelvin, 1/picosecond, 0.01*picoseconds)
      5 

~/miniconda3/envs/antibody/lib/python3.9/site-packages/openmm/app/forcefield.py in createSystem(self, topology, nonbondedMethod, nonbondedCutoff, constraints, rigidWater, removeCMMotion, hydrogenMass, residueTemplates, ignoreExternalBonds, switchDistance, flexibleConstraints, drudeMass, **args)
   1204         # Find the template matching each residue and assign atom types.
   1205 
-> 1206         templateForResidue = self._matchAllResiduesToTemplates(data, topology, residueTemplates, ignoreExternalBonds)
   1207         for res in topology.residues():
   1208             if res.name == 'HOH':

~/miniconda3/envs/antibody/lib/python3.9/site-packages/openmm/app/forcefield.py in _matchAllResiduesToTemplates(self, data, topology, residueTemplates, ignoreExternalBonds, ignoreExtraParticles, recordParameters)
   1418                             break
   1419             if matches is None:
-> 1420                 raise ValueError('No template found for residue %d (%s).  %s' % (res.index+1, res.name, _findMatchErrors(self, res)))
   1421             else:
   1422                 if recordParameters:

ValueError: No template found for residue 30730 (HOH).  The set of atoms matches HOH, but the bonds are different.
```

I tried many forcefileds. They threw errors as well, while the residue num differs.
I also checked the water residues. They seem good, however.
```
residues = list(pdb.topology.residues())
list(residues[30730].atoms())
```
the output:
```
[<Atom 100330 (O) of chain 3 residue 30730 (HOH)>,
 <Atom 100331 (H1) of chain 3 residue 30730 (HOH)>,
 <Atom 100332 (H2) of chain 3 residue 30730 (HOH)>]
```
`list(residues[30730].bonds())`
the output:
```
[Bond(<Atom 100331 (H1) of chain 3 residue 30730 (HOH)>, <Atom 100330 (O) of chain 3 residue 30730 (HOH)>),
 Bond(<Atom 100332 (H2) of chain 3 residue 30730 (HOH)>, <Atom 100330 (O) of chain 3 residue 30730 (HOH)>)]
```
I uploaded the input pdb and solvated ones for repeat.
[error.zip](https://github.com/openmm/openmm/files/7773555/error.zip)
Thanks in advance.

**Comment by pengtb** [comment]
Using `modeller.topology` directly to create a system succeeds. Maybe there's sth wrong with PDBFile IO?

**Comment by peastman** [comment]
Sorry for the delay in replying.  I'm just catching up after the holidays.  I thought I had already replied to this, but it looks like I didn't.

The PDB format doesn't support models with more than 100,000 atoms.  It only provides five columns for the atom number.  If you have more than that, the numbers wrap around and start repeating.  Then it gets confused by the CONECT records describing bonds within the water molecules, because they specify atoms by number, and there are multiple atoms with the same number.

I recommended using PDBxFile to write to a PDBx/mmCIF file instead.  That format is the successor to PDB, which is now considered obsolete.
